src/stream/src/task/barrier_manager.rs

```diff
@@ -92,34 +92,33 @@ enum LocalBarrierEvent {
 /// barriers to and collect them from all actors, and finally report the progress.
 struct LocalBarrierWorker {
     /// Stores all streaming job source sender.
-    senders: HashMap<ActorId, Vec<UnboundedSender<Barrier>>>,
+    barrier_senders: HashMap<ActorId, Vec<UnboundedSender<Barrier>>>,
 
     /// Current barrier collection state.
     state: ManagedBarrierState,
 
     /// Save collect `CompleteReceiver`.
     collect_complete_receiver: HashMap<u64, CompleteReceiver>,
 
-    streaming_metrics: Arc<StreamingMetrics>,
+    /// Record all unexpected exited actors.
+    failure_actors: HashMap<ActorId, StreamError>,
```

Comment:
any reason we wanna keep this?

Reply:
Just to be consistent with the original logic. We can save the error and early return the error when handling inject_barrier.